device/dell/x86_64-dell_s6100_c2538-r0/Force10-S6100/buffers.json.j2

```diff
@@ -35,7 +35,25 @@
     {%- if cable_len -%}
         {{ cable_len.0 }}
     {%- else -%}
-        {{ default_cable }}
+        {%- if switch_role == 'ToRRouter' -%}
+            {%- for local_port in VLAN_MEMBER -%}
+                {%- set vlan_port = local_port.split("|") -%}
+                {%- if vlan_port[1] == port_name -%}
+                    {%- set roles3 = switch_role + '_' + 'server' -%}
```

Comment:
I think you should check neighbor role first, acting accordingly, if the neighbor role is missing, then default to use vlan member v.s. non-vlan member, assuming non-vlan members are up links.

Can you elaborate why the vlan member were tested first?

Reply:
@yxieca As I explained to you the if check earlier takes care of the neighbor role and based on the neighbor role the cable length is taken. For the server side ports if the neighbor role is missing then we come to the else part where we check for the vlan members for those ports and get the correct cable length.